daemon/src/collab_settlement_maker.rs

```diff
@@ -0,0 +1,212 @@
+    proposal: SettlementProposal,
+    taker_id: Identity,
+    connections: Box<dyn MessageChannel<maker_inc_connections::settlement::Response>>,
+    on_stopping: Vec<Box<dyn MessageChannel<Stopping<Self>>>>,
+}
+
+pub enum Completed {
+    Confirmed {
+        order_id: OrderId,
+        settlement: CollaborativeSettlement,
+        script_pubkey: Script,
+    },
+    Rejected {
+        order_id: OrderId,
+    },
+    Failed {
+        order_id: OrderId,
+        error: anyhow::Error,
+    },
+}
+
+pub struct Accepted;
+pub struct Rejected;
+pub struct Initiated {
+    pub sig_taker: Signature,
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle(&mut self, _: Accepted, ctx: &mut xtra::Context<Self>) {
+        let order_id = self.cfd.order.id;
+
+        tracing::info!(%order_id, "Settlement proposal accepted");
+
+        self.accept(ctx).await;
+        self.update_proposal(None).await;
+    }
+
+    async fn handle(&mut self, _: Rejected, ctx: &mut xtra::Context<Self>) {
+        let order_id = self.cfd.order.id;
+
+        tracing::info!(%order_id, "Settlement proposal rejected");
+
+        self.reject(ctx).await;
+        self.update_proposal(None).await;
+    }
+
+    async fn handle(&mut self, msg: Initiated, ctx: &mut xtra::Context<Self>) {
+        let completed = async {
+            tracing::info!(
+                order_id = %self.cfd.order.id,
+                taker_id = %self.taker_id,
+                "Received signature for collaborative settlement"
+            );
+
+            let Initiated { sig_taker } = msg;
+
+            let dlc = self.cfd.open_dlc().context("CFD was in wrong state")?;
+            let (tx, sig_maker) = dlc.close_transaction(&self.proposal)?;
+            let spend_tx = dlc.finalize_spend_transaction((tx, sig_maker), sig_taker)?;
+
+            let settlement = CollaborativeSettlement::new(
+                spend_tx.clone(),
+                dlc.script_pubkey_for(Role::Maker),
+                self.proposal.price,
+            )?;
+
+            self.update_proposal(None).await;
+
+            anyhow::Ok(Completed::Confirmed {
+                order_id: self.cfd.order.id,
+                settlement,
+                script_pubkey: dlc.script_pubkey_for(Role::Maker),
+            })
+        }
+        .await
+        .unwrap_or_else(|e| Completed::Failed {
+            order_id: self.cfd.order.id,
+            error: e,
+        });
+
+        self.complete(completed, ctx).await;
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Actor {
+    async fn started(&mut self, _ctx: &mut xtra::Context<Self>) {
+        tracing::info!(
+            order_id = %self.proposal.order_id,
+            price = %self.proposal.price,
+            "Received settlement proposal"
+        );
+
+        self.update_proposal(Some((self.proposal.clone(), SettlementKind::Incoming)))
+            .await;
+
+        todo!("Update maker_inc_connections::Actor with our address");
```

Comment:
@luckysori We need to deal with this one :D

Reply:
Ah, forgot to remove it. We should be registered as soon as the user accepts.